syntax = "proto3";

option go_package = "github.com/xyberii4/meepo/gateway/internal/grpc/pb";

service BotService {
  rpc JoinMeeting (JoinMeetingRequest) returns (stream JoinMeetingResponse);
  rpc GetMeetingDetails (MeetingDetailsRequest) returns (MeetingDetailsResponse);
  rpc LeaveMeeting (LeaveMeetingRequest) returns (LeaveMeetingResponse);
}

message JoinMeetingRequest {
  string meepo_id = 1;
  string bot_id = 2;
  string url = 3;
  string name = 4;
}

message JoinMeetingResponse {
  enum State {
    RECEIVED = 0;
    PENDING = 1;
    JOINED = 2;
    FAILED = 3;
  }

  State state = 1;

  string message = 2;
  string bot_id = 3;
}

message MeetingDetailsRequest {
  string bot_id = 1;
  string meepo_id = 2;
}

message MeetingDetailsResponse {
  repeated Participant participants = 1;
}

message Participant {
  string name = 1;
}

message LeaveMeetingRequest {
  string bot_id = 1;
  string meepo_id = 2;
}

message LeaveMeetingResponse {
  enum State {
    RECEIVED = 0;
    DONE = 1;
    FAILED = 2;
  }

  State state = 1;

  string message = 2;
}
